ecosystem/node-checker/src/metric_evaluator/consensus/proposals_evaluator.rs

```diff
@@ -0,0 +1,214 @@
+                    headline: "Proposals count is not progressing".to_string(),
+                    score: 50,
+                    explanation: "Successfully pulled metrics from target node twice, but the proposal count isn't progressing.".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+              }
+            }
+            progress if (progress < 0) => {
+                EvaluationResult {
+                    headline: "Proposals count went backwards!".to_string(),
+                    score: 0,
+                    explanation: "Successfully pulled metrics from target node twice, but the second time the proposals count went backwards!".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+                }
+            }
+            _wildcard => {
+                EvaluationResult {
+                    headline: "Proposals count is increasing".to_string(),
+                    score: 100,
+                    explanation: "Successfully pulled metrics from target node twice and saw that proposals count is increasing".to_string(),
+                    source: CONSENSUS_EVALUATOR_SOURCE.to_string(),
+                    links: vec![],
+                }
+            }
+        }
+    }
+}
+
+impl MetricsEvaluator for ConsensusProposalsEvaluator {
+    /// Assert that the proposals count is increasing on the target node.
+    fn evaluate_metrics(
+        &self,
+        _previous_baseline_metrics: &PrometheusScrape,
+        previous_target_metrics: &PrometheusScrape,
+        _latest_baseline_metrics: &PrometheusScrape,
+        latest_target_metrics: &PrometheusScrape,
+    ) -> Result<Vec<EvaluationResult>, MetricsEvaluatorError> {
+        let mut evaluation_results = vec![];
+
+        // Get previous proposals count from the target node.
+        let previous_proposals_count =
+            match self.get_proposals_count(previous_target_metrics, "first") {
+                GetMetricResult::Present(metric) => Some(metric),
+                GetMetricResult::Missing(evaluation_result) => {
+                    evaluation_results.push(evaluation_result);
+                    None
+                }
+            };
+
+        // Get the latest proposals count from the target node.
+        let latest_proposals_count = match self.get_proposals_count(latest_target_metrics, "second")
+        {
+            GetMetricResult::Present(metric) => Some(metric),
+            GetMetricResult::Missing(evaluation_result) => {
+                evaluation_results.push(evaluation_result);
+                None
+            }
+        };
+
+        match (previous_proposals_count, latest_proposals_count) {
+            (Some(previous), Some(latest)) => {
+                evaluation_results.push(self.build_evaluation(previous, latest));
+            }
+            _ => {
+                debug!(
```

Comment:
nit: this feels like a `warn` level log to me 😄

Reply:
I figure only things wrong with our own setup are worth warn+, if a target node has some issue that doesn't warrant a warning on our end imo.